src/storage/mvcc/reader.rs

```diff
@@ -194,34 +194,33 @@ impl MvccReader {
         Ok(Some((commit_ts, write)))
     }
 
-    fn check_lock(&mut self, key: &Key, mut ts: u64) -> Result<Option<u64>> {
+    fn check_lock(&mut self, key: &Key, mut ts: u64) -> Result<()> {
         if let Some(lock) = self.load_lock(key)? {
-            if lock.ts <= ts {
-                if ts == u64::MAX && key.raw()? == lock.primary {
-                    // when ts==u64::MAX(which means to get latest committed version for
-                    // primary key),and current key is the primary key, returns the latest
-                    // commit version's value
-                    ts = lock.ts - 1;
-                } else {
-                    // There is a pending lock. Client should wait or clean it.
-                    return Err(Error::KeyIsLocked {
-                        key: key.raw()?,
-                        primary: lock.primary,
-                        ts: lock.ts,
-                        ttl: lock.ttl,
-                    });
-                }
+            if lock.ts > ts ||
+                lock.lock_type == LockType::Lock ||
+                ts == u64::MAX && key.raw()? == lock.primary {
```

Comment:
Do one thing in one PR? I do not think is a good idea to change (ts==u64::MAX)'s logic in this PR.

Reply:
Just make the logic consistent.